src/base/bittorrent/torrentimpl.cpp

```diff
@@ -937,6 +938,21 @@ bool TorrentImpl::addTag(const Tag &tag)
     return true;
 }
 
+bool TorrentImpl::setTags(const TagSet &newTags)
+{
+    // Identify tags to add
+    for (const Tag &tag : newTags)
+        if (!hasTag(tag))
+            addTag(tag);
+
+    // Identify tags to remove
+    for (const Tag &tag : asConst(tags()))
+        if (!newTags.contains(tag))
+            removeTag(tag);
+
+    return true;
```

Comment:
why does it need to return a bool?

Reply:
No good reason other than the above methods does it. Should it be a void instead?

Reply:
yea I think